components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -4869,22 +4896,30 @@ public Response reimportServiceFromCatalog(String apiId, MessageContext messageC
             } else if (ServiceEntry.DefinitionType.ASYNC_API.equals(service.getDefinitionType())) {
                 validationResponseMap = validateAsyncAPISpecification(null, service.getEndpointDef(),
                         null, true, true);
-            } else {
+            } else if (!ServiceEntry.DefinitionType.WSDL1.equals(service.getDefinitionType())) {
```

Comment:
If something does not fall into any of the above categories, I think we might need an else condition to handle that. WDYT?

Reply:
Yes, this is kind of else consition because it gets all other categories except WSDL1 types.